test/SparqlAntlrParserTestHelpers.h

```diff
@@ -132,39 +136,32 @@ constexpr const ad_utility::Last<Current, Others...>* unwrapVariant(
  * @param resultOfParseAndText Parsing result
  * @param matcher Matcher that must be fulfilled
  */
-void expectCompleteParse(const auto& resultOfParseAndText, auto&& matcher) {
+void expectCompleteParse(
+    const auto& resultOfParseAndText, auto&& matcher,
+    std::source_location l = std::source_location::current()) {
+  testing::ScopedTrace trace(l.file_name(), l.line(), "Source location");
   EXPECT_THAT(resultOfParseAndText.resultOfParse_, matcher);
   EXPECT_TRUE(resultOfParseAndText.remainingText_.empty());
 }
 
 // _____________________________________________________________________________
-/**
- * Ensures that resultOfParseAndText.resultOfParse_ is an array-like type (e.g.
- * std::vector) the size of which is the number of specified matchers and the
- * the i-th matcher matches resultOfParseAndText.resultOfParser_[i] and that the
- * text has been fully consumed by the parser.
- *
- * @param resultOfParseAndText Parsing result
- * @param matchers Matcher... that must be fulfilled
- */
-void expectCompleteArrayParse(const auto& resultOfParseAndText,
-                              auto&&... matchers) {
-  auto expect_single_element = [](auto&& result, auto matcher) {
-    EXPECT_THAT(result, matcher);
-  };
-  ASSERT_EQ(resultOfParseAndText.resultOfParse_.size(), sizeof...(matchers));
-  auto sequence = std::make_index_sequence<sizeof...(matchers)>();
-
-  auto lambda = [&]<size_t... i>(std::index_sequence<i...>) {
-    (...,
-     expect_single_element(resultOfParseAndText.resultOfParse_[i], matchers));
-  };
-
-  lambda(sequence);
-  EXPECT_TRUE(resultOfParseAndText.remainingText_.empty());
+void expectIncompleteParse(const auto& resultOfParseAndText, const string& rest,
+                           auto&& matcher) {
+  EXPECT_THAT(resultOfParseAndText.resultOfParse_, matcher);
+  EXPECT_THAT(resultOfParseAndText.remainingText_, testing::StrEq(rest));
+}
+[[maybe_unused]] void expectIncompleteParse(const auto& resultOfParseAndText,
+                                            const size_t length,
+                                            auto&& matcher) {
+  EXPECT_THAT(resultOfParseAndText.resultOfParse_, matcher);
+  EXPECT_THAT(resultOfParseAndText.remainingText_, testing::SizeIs(length));
+}
+void expectIncompleteParse(const auto& resultOfParseAndText, auto&& matcher) {
+  expectIncompleteParse(resultOfParseAndText, 0, matcher);
 }
-// _____________________________________________________________________________
 
+// Cannot be broken down into a combination of existing googletest matchers
+// because unwrapVariant is not available as a matcher.
```

Comment:
The comment is fine for now, 
but I think in a small additional PR we can implement `unwrapVariant` s.t. it takes a matcher and returns a matcher, then we are done here:).